### PR TITLE
Converting tuple to namedtuple for editor method - issue #6004

### PR DESCRIPTION
**Pull request by amentee**
## What does this PR change?

It converts tuple to namedtuple for editor method in spacecmd/src/spacecmd/utils.py file.

## GUI diff

No difference.

Before:

After:

- [ ] **DONE**

## Documentation
- No documentation needed: **add explanation. This can't be used if there is a GUI diff**
- No documentation needed: only internal and user invisible changes
- Documentation issue was created: [Link for SUSE Manager contributors](https://github.com/SUSE/spacewalk/issues/new?template=ISSUE_TEMPLATE_DOCUMENTATION.md&labels=documentation&projects=SUSE/spacewalk/31), [Link for community contributors](https://github.com/uyuni-project/uyuni-docs/issues/new).
- API documentation added: please review the Wiki page [Writing Documentation for the API](https://github.com/uyuni-project/uyuni/wiki/Writing-documentation-for-the-API) if you have any changes to API documentation.
- (OPTIONAL) [Documentation PR](https://github.com/uyuni-project/uyuni-docs/pulls)

- [ ] **DONE**

## Test coverage
- No tests: **add explanation**
- No tests: already covered
- Unit tests were added
- Cucumber tests were added

- [ ] **DONE**

## Links

Fixes #
Tracks # **add downstream PR, if any**

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by amentee** [comment]
@agraul - Please review and let me know if this is good or needs some changes

File: spacecmd/src/spacecmd/utils.py [hunk 1]
```diff
@@ -244,6 +245,7 @@ def editor(template='', delete=False):
             handle = open(file_name, 'r')
             contents = handle.read()
             handle.close()
+            File = namedtuple("file", ("contents", "file_name"))
```
Comment (agraul) [inline spacecmd/src/spacecmd/utils.py:248]: Normally, type declarations are done in the module scope. Is there a specific reason you've opted to have it inside the function?

File: spacecmd/src/spacecmd/utils.py [hunk 2]
```diff
@@ -252,7 +254,7 @@ def editor(template='', delete=False):
                 except OSError:
                     logging.error(_N('Could not remove %s'), file_name)
 
-            return contents, file_name
+            return File(contents, file_name)
         except IOError:
             logging.error(_N('Could not read %s'), file_name)
             return [], ''
```
Comment (agraul) [inline spacecmd/src/spacecmd/utils.py:260]: I think this return statement should use the same type.

**Review by agraul — CHANGES_REQUESTED** [review]
Thanks a lot for the pull request! I left two comments inline, please take a look when you have time.

The idea behind #6004 is to use named tuples everywhere in `spacecmd/src/` where currently tuples are returned, the `editor()` function is just an example to get you started. 

It would be great if you could check more usages where tuples are currently returned and come up with meaningful names for the fields.

**Comment by Etheryte** [comment]
@agraul Is this still relevant or should we close this as stale?

**Comment by agraul** [comment]
> @agraul Is this still relevant or should we close this as stale?

Let's close it for now. @amentee if you want to continue working on this PR, please let us know.

**Pull request closed without merging**